src/utils/update/trusts.ts

```diff
@@ -0,0 +1,12 @@
+import { ApplicationData } from "model";
+
+export const checkHasAnyBosWithTrusteeNocs = (appData: ApplicationData) => {
+  const bosThatCanHaveTrusteeNoc = [
```

Comment:
quite a bit of work to join the lists into one big one just to check some. Could just use 
[appData.beneficial_owners_individual, appData.beneficial_owners_corporate].some(l => l.map(bo => bo.trustees_nature_of_control_types > 0));

Reply:
I don't see the benefit of flatMap here either. Can you write up a suggestion? 

Tbh I think `[...(appData.array || []), ...]` is the simplest way of flattening those arrays.

Reply:
[ appData.array1, appData.array2].flat().some(a => a?.length > 0);

was hoping this handles undefines so no need for "|| []"

Reply:
typescript flat() does not seem to deal well with undefines. Some languages just filter those. So how about the some of some suggestion. Just to iterate over the list - not expand everything into a new list as [ ...array1, ...array2 ] does:

type X = {
   t? : Array<String>;
};

const l0 : Array<X> = [];
const l1 : Array<X> = [ {} ];
const l2 : Array<X> | undefined = undefined;
const l3 : Array<X> = [{ t: ["here "] }];

const result = [l0, l1, l2, l3].some(x => (x ?? []).some(y => ( y.t ?? []).length > 0));

console.log(result);

Reply:
I don't think this solution is as readable, and if there's an implied argument for potential performance improvements, I'd expect the benefit to be negligible.

Gonna make a call here and say I don't think this change would provide any real benefit.

Reply:
There is duplication with scr/utils/trust.ts which checks for NoC too.